client/components/domains/map-domain-step/index.jsx

```diff
@@ -255,6 +255,11 @@ const MapDomainStep = React.createClass( {
 				message = translate( 'Please enter a domain name or keyword.' );
 				break;
 
+			case 'recently_mapped':
+				message = translate( 'This domain was recently in use by someone else and is not available to map yet. ' +
+					'Please try again later or contact support.' );
+				break;
```

Comment:
Feel free to ignore my nitpick but maybe we could format this with the string on a new line so that they're aligned nicely
```js
    message = translate(
        'This domain was recently in use by someone else and is not available to map yet. ' +
        'Please try again later or contact support.'
    );
    break;
```